src/index_notation/provenance_graph.cpp

```diff
@@ -461,6 +473,23 @@ std::vector<ir::Expr> PosRelNode::deriveIterBounds(taco::IndexVar indexVar,
                                                      std::map<taco::IndexVar, taco::ir::Expr> variableNames,
                                                      Iterators iterators,
                                                      ProvenanceGraph provGraph) const {
+                                                      
+  
+  if (indexVar.isBound()){
```

Comment:
In the updated pull request, the provenance graph keeps a map of bounded vars. We cannot add the bound as VarRelNode instead since we want to get around the problem of having to create a new indexVar in order to bound it. 

At it's current implementation, we cannot have cycles in the provenance graph, but the bound (if implemented as a RelNode, we will introduce a cycle since the parent and child would be the same).